src/functions.php

```diff
@@ -40,4 +42,56 @@ function displaySocksCollection(PDO $db): string {
         echo 'not working';
     }
     return createSockDiv($socksArr);
+}
+
+function getTable(string $table,PDO $db): array {
+    $query = $db->prepare("SELECT `{$table}` FROM `{$table}s`");
+    $result = $query->execute();
+    if ($result) {
+        return $query->fetchAll();
+    } else {
+        throw new Exception('error');
+    }
+}
+
+function dropDownOptions(array $rows): string {
+    $options= '';
+    foreach($rows as $row) {
+        foreach ($row as $key => $value) {
+            $options .= "<option value=\"{$value}\">{$value}</option>";
+        }
+    }
+    return $options;
+}
+
+function createDropdown(string $name) : string {
+    $start = "<select name=\"{$name}\" id=\"{$name}\"><option disabled=\"disabled\" selected=\"selected\">$name</option>";
+    $options = dropDownOptions(getTable($name, connectDB()));
+    $end = '</select>';
+    return $start . $options . $end;
+}
+
+if ($_GET) {
+    $db = connectDB();
+    $query = $db->prepare("INSERT INTO `socks` (`name`, `size`, `pattern`, `color`, `description`) VALUES (:name, :size, :pattern, :color, :description)");
+    //$query = $db->prepare("INSERT INTO `socks` (`size`, `pattern`, `color`) VALUES (:size, :pattern, :color)");
+    $result = $query->execute([
+        'name' => $_GET['name'],
+        'size' => getRelatedNumberForDropdownOption('size', $_GET['size']),
+        'pattern' => getRelatedNumberForDropdownOption('pattern', $_GET['pattern']),
+        'color' => getRelatedNumberForDropdownOption('color', $_GET['color']),
+        'description' => $_GET['description']
+    ]);
+}
```

Comment:
This is not in a function - it only accidentally works because you `require` the file it is in. More thinking needed.